packages/lib/src/vis/shared/Html.tsx

```diff
@@ -1,145 +1,38 @@
-  const group = useRef<Group>(null!); // eslint-disable-line @typescript-eslint/no-non-null-assertion
-
-  const getGroupPosition = useCallback(() => {
-    const worldPos = new Vector3().setFromMatrixPosition(
-      group.current.matrixWorld
-    );
-    const htmlPos = worldToHtml(worldPos);
-
-    return [htmlPos.x, htmlPos.y];
-  }, [worldToHtml]);
+  const { container = gl.domElement.parentElement, children } = props;
 
-  const getGroupScale = useCallback(() => {
-    const groupScale = new Vector3().setFromMatrixScale(
-      camera.matrixWorldInverse
-    ); // Scale inverse to the camera scale
+  const [renderTarget] = useState(() => document.createElement('div'));
 
-    return [groupScale.x, groupScale.y];
-  }, [camera]);
-
-  const getInnerDivTransform = useCallback(() => {
-    const position = followCamera ? getGroupPosition() : undefined;
-    const scale = scaleOnZoom ? getGroupScale() : undefined;
-
-    return [
-      position && `translate3d(${position[0]}px, ${position[1]}px, 0)`,
-      scale && `scale(${scale[0]}, ${scale[1]})`,
-    ]
-      .filter(isDefined)
-      .join(' ');
-  }, [followCamera, getGroupPosition, getGroupScale, scaleOnZoom]);
-
-  // Append/remove container `div` next to R3F's `canvas`
   useLayoutEffect(() => {
-    if (parentElement) {
-      parentElement.append(el);
-    }
+    ReactDOM.render(<>{children}</>, renderTarget); // eslint-disable-line react/jsx-no-useless-fragment
+  }, [children, renderTarget]);
 
+  useLayoutEffect(() => {
     return () => {
-      if (parentElement) {
-        el.remove();
-      }
-
-      ReactDOM.unmountComponentAtNode(el);
+      ReactDOM.unmountComponentAtNode(renderTarget);
     };
-  }, [el, parentElement]);
+  }, [renderTarget]);
 
-  // Update size and overflow of container `div`
   useLayoutEffect(() => {
-    el.style.width = `${width}px`;
-    el.style.height = `${height}px`;
-    el.style.overflow = followCamera ? 'hidden' : 'visible';
-  }, [el, width, height, followCamera]);
+    container?.append(renderTarget);
 
-  // Render inner `div`
-  useEffect(() => {
-    group.current.updateMatrixWorld(); // make sure we get the group's most up to date position
-
-    ReactDOM.render(
-      <div
-        ref={innerRef}
-        className={className}
-        style={{
-          position: 'absolute',
-          top: 0,
-          left: 0,
-          transformOrigin: 'left top',
-          transform: getInnerDivTransform(),
-          ...style,
-        }}
-        {...divProps}
-      >
-        {children}
-      </div>,
-      el
-    );
-  }, [children, className, divProps, el, getInnerDivTransform, style]);
-
-  // Update inner `div` position on every frame
-  useFrame(() => {
-    if (innerRef.current) {
-      innerRef.current.style.transform = getInnerDivTransform();
-    }
-  });
+    return () => {
+      renderTarget.remove();
+    };
+  }, [container, renderTarget]);
 
-  return <group ref={group} {...groupProps} />;
+  return null;
```

Comment:
Really dumb question: I guess returning `createPortal(children, container)` doesn't work as it gets picked up by R3F renderer and not ReactDOM's ?

Reply:
Exactly! R3F does have its own `createPortal`, but the problem is also with the `children`, which contain HTML markup instead of R3F markup.